tensorflow_addons/optimizers/moving_average.py

```diff
@@ -124,3 +124,57 @@ def _create_slots(self, var_list):
         )  # pylint: disable=protected-access
         for var in var_list:
             self.add_slot(var, "average", var.read_value())
+
+    def shadow_copy(self, model_weights):
+        """Creates shadow variables for the given model weights."""
+        for var in model_weights:
+            self.add_slot(var, "average", initializer="zeros")
+        self._average_weights = [self.get_slot(var, "average") for var in model_weights]
+        self._model_weights = model_weights
+
+    @property
+    def has_shadow_copy(self):
+        """Whether this optimizer has created shadow variables."""
+        return self._model_weights is not None
+
+    def swap_weights(self):
+        """Swap the average and moving weights.
+
+      This is a convenience method to allow one to evaluate the averaged weights
+      at test time. Loads the weights stored in `self._average` into the model,
+      keeping a copy of the original model weights. Swapping twice will return
+      the original weights.
+      """
+        if tf.distribute.in_cross_replica_context():
+            strategy = tf.distribute.get_strategy()
+            return strategy.experimental_run_v2(self._swap_weights, args=())
```

Comment:
This function is called here and in the test case. 

I know experimental and private apis aren't typically allowed in tf_addons because they have a high maintenance cost. 

However, this function has been updated to no longer be experimental: https://github.com/tensorflow/tensorflow/commit/0b8f0a5b84852bd0ac536c3c58c32793be13ba2a

What is your policy for cases like this?

Reply:
I believe we should wait until it's not experimental anymore and then we can use the public API :)

Reply:
@gabrieldemarmiesse So this API is public on the 2.2 branch. Some possible options:
* Raise our minimum TF version if we accept this so 0.10 is >= TF2.2 compatible
* Do a not so elegant version check to use experimental_run_v2 for TF2.1

Reply:
I'd be in favor of dropping tf 2.1 then. I believe there is another pull request that needs tf 2.2 to use a feature that was experimental before.

Reply:
@tf-marissaw would you mind updating the PR to use the public `distribution.run` API in TF2.2. We will mark this PR as blocked until we adjust the CI and prepare a  >= TF 2.2 nightly and future release (should be a few days after 2.2 release)

Reply:
We've switch to mastering only building TF2.2 so this is unblocked. Please update the PR when time allows... thx!

Reply:
Will do, thanks!

Reply:
Done